### PR TITLE
Proposal to accept NAP-1 - add institutional and funding partners to governance

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
Following up on NAP 1 #4446 this PR adds the proposed change to our governance. Discussion can happen here or on [zulip](https://napari.zulipchat.com/#narrow/stream/322105-naps/topic/nap-1!/near/280631529) and the actual NAP will be updated accordingly to capture any discussion for the record. Thanks!!

### CODE REVIEW

**Comment by sofroniewn** [comment]
Also I have marked this PR as draft so it does not accidentally get merged, but it is in fact ready for review and comments from @napari/core-devs

Additionally I should add that if this PR is accepted then there will be a proposal to the steering council to make CZI our first institutional and funding partner. That would happen in a follow up PR, but I just wanted to let folks know now for full transparency. cc @jni

**Comment by jni** [comment]
Thanks @sofroniewn!

I would potentially rename this to: "Proposal to accept NAP-1" and start a thread about it on the [#naps channel on Zulip](https://napari.zulipchat.com/#narrow/stream/322105-naps/topic/nap-1!/near/280631529), and mark the NAP itself as "Accepted".

That can wait until tomorrow but no longer if we want to stay within the 7 day minimum discussion time for NAP acceptance.

File: docs/community/governance.md
Comment (brisvag) [inline docs/community/governance.md:87]: Does this mean that a non-core developer who joins the SC will also automatically become a core dev?

File: docs/community/governance.md
Comment (brisvag) [inline docs/community/governance.md:101]: What about abbreviating to IFP to make the following text a bit less unwieldy? I feel like half of my time was spent reading "Institutional and Funding Partners" xD Also, SC is sometimes abbreviated, sometimes not.

File: docs/community/governance.md
Comment (brisvag) [inline docs/community/governance.md]: Should this document include instructions on how to deal with (or identify) situations where this is *not* the case?

File: docs/community/governance.md
Comment (brisvag) [inline docs/community/governance.md]: This will happen immediately, or on the following january?

File: docs/community/governance.md
Comment (jni) [reply docs/community/governance.md:87]: no. They are merely the IFP SC member. This line is just to contrast it with the core-dev -> SC path.

File: docs/community/governance.md
Comment (jni) [reply docs/community/governance.md]: :grimacing: Dunno. :joy: Basically, I feel like putting in those instructions breaks the "good faith" "spell". But what I envision is that since IFP status is decided by the SC, it can also be revoked by the SC. So... :shrug: :grimacing: Do you have harsher punishments in mind? :joy:

File: docs/community/governance.md
Comment (jni) [reply docs/community/governance.md]: I would argue immediately. I read the January thing is more of a prod to think about turning over, rather than as a date where *all* SC transactions have to happen.

File: docs/community/governance.md
Comment (brisvag) [reply docs/community/governance.md]: Ahah, no, revoking seems about right :P I'm mostly thinking about concerns of the community, rather than the SC itself; I feel like it would be useful to have something concrete to *point* to if, say, someone who's not directly in contact with the SC/core-devs wants to raise a concern to the SC about an IFP not acting in "good faith". Of course, this is not as easy as a checklist of parameters that make some officially "good faith" :P But yeah, I'm just thinking out loud, this line might already be what I'm looking for :)

File: docs/community/governance.md
Comment (brisvag) [reply docs/community/governance.md]:
```suggestion
will cease being a member of the SC and the remaining Institutional and
Funding Partner Advisory Council members may choose a new Representative
at their earliest convenience.
```

File: docs/community/governance.md
Comment (alisterburt) [inline docs/community/governance.md]: not particularly concerned but worth highlighting: the possibility of an SC of four members is now very real, the fact that this could cause problems around decision making is raised but no mechanisms are defined for how to deal with this issue if it arises

edit: I trust the SC to be able to manage this which is why I'm not particularly concerned :-)

File: docs/community/governance.md
Comment (alisterburt) [inline docs/community/governance.md]: highlighting a potential side benefit of this: this provides a reason for employers of existing napari contributors/core-devs to officially recognise work on napari as part of someones 'duties', potentially legitimising work on OSS more generally as 'useful work' 🥳

File: docs/community/governance.md
Comment (alisterburt) [inline docs/community/governance.md]: given that the SC have no oversight here, the SC/wider community should probably review these terms during the instantiation of the first IFP to ensure everything seems reasonable from the outset 🙂

**Review by alisterburt — APPROVED** [review]
thanks for all of the thought/work that has gone into this - establishing the interface between napari and larger institutions wanting to provide support should help ensure long term success of the project 🙂 

Few minor comments/thoughts below, looking forward to discussing in SF!

File: docs/community/governance.md
Comment (brisvag) [reply docs/community/governance.md]: Right; should we require/expect such organization to happen somewhere in the open?

File: docs/community/governance.md
Comment (neuromusic) [inline docs/community/governance.md:63]: Both words should be capitalized here, yes? (since it is a proper noun)

File: docs/community/governance.md
Comment (neuromusic) [inline docs/community/governance.md]:
```suggestion
The napari Steering Council (SC) will be no less than three members and no
```

File: docs/community/governance.md
Comment (neuromusic) [inline docs/community/governance.md]:
```suggestion
employed by the same entity. One seat on the Steering Council is reserved for
```

File: docs/community/governance.md
Comment (jni) [reply docs/community/governance.md]: I *hope* that the SC is always listening to the community! :grimacing: :pray:

File: docs/community/governance.md
Comment (jni) [reply docs/community/governance.md]: :shrug: I dunno. I actually added this explicitly because I'd like to not be concerned here — if they want to make their decisions through gladiatorial-style combat, that's fine :joy:, as long as they send us a representative that actually represents them. :joy: My *only* concern is if some orgs wield their power and influence too haphazardly, other orgs might feel disenfranchised. But that's the "good faith" clause and I hope that as long as we have various communications channels (and there *is* an IFP council and the SC will meet with that council regularly), then we should be ok?

But, the idea of mandating transparency is a good one. Do either of you want to suggest some wording around it?

Comment (brisvag) [reply docs/community/governance.md]: As long as the gladiatorial combat is open to everyone I'm on board 😆 I'm not sure... we could just have a zulip channel dedicated to such discussions, though probably face-to-face meetings are better for this sort of decision.

Comment (sofroniewn) [reply docs/community/governance.md]: I think having the SC review when the first IFP comes on is a good idea - i'm not sure that needs to be formally captured in governance as it will only happen once. I'm imagining that a good-faith approach would somehow incorporate the amount of support each partner is giving - but is it exactly a weighted sum of employees and grant dollars or something a bit more nuanced I think can be deferred to later

File: docs/community/governance.md
Comment (justinelarsen) [reply docs/community/governance.md]: I agree with @alisterburt 's concern, but I have faith in the SC to work it out. Perhaps some wording on what to do in the rare case that an agreement can't be made amongst the SC members would be helpful for the future.

Also, I'm curious about the limit of five members? I think five is a great number and makes sense, but what about seven? Is there a particular reason for the limit at five?

File: docs/community/governance.md
Comment (justinelarsen) [reply docs/community/governance.md]: Is there only one "Institutional and Funding Partner Advisory Council" which all of the IFPs contribute to? Or does each IFP have their own Advisory Council and each elects a representative to sit on the SC?

**Review by justinelarsen — APPROVED** [review]
Great proposal and planning ahead, thanks for creating this!

File: docs/community/governance.md
Comment (sofroniewn) [reply docs/community/governance.md:101]: Added. I have tried to improve

File: docs/community/governance.md
Comment (sofroniewn) [reply docs/community/governance.md]: added to discussion

File: docs/community/governance.md
Comment (sofroniewn) [reply docs/community/governance.md:87]: added to discussion

File: docs/community/governance.md
Comment (sofroniewn) [reply docs/community/governance.md]: suggestion added

File: docs/community/governance.md
Comment (sofroniewn) [reply docs/community/governance.md]: > Perhaps some wording on what to do in the rare case that an agreement can't be made amongst the SC members would be helpful for the future.

I have added a line that if the vote is deadlocked it will be deferred until an odd number of members is present.

> Is there a particular reason for the limit at five?

It was thought to be a good limit with one IFP seat. The open forcefield project has 7 members with two IFP seats.

File: docs/community/governance.md
Comment (sofroniewn) [reply docs/community/governance.md]: Yes there is only one IFP Advisory council with a single representative from each IFP. They are expected to elect that single seat for the SC

Comment (sofroniewn) [reply docs/community/governance.md]: > But, the idea of mandating transparency is a good one. Do either of you want to suggest some wording around it?

I have added something

File: docs/community/governance.md
Comment (sofroniewn) [reply docs/community/governance.md]: fixed

File: docs/community/governance.md
Comment (sofroniewn) [reply docs/community/governance.md]: fixed

**Comment by sofroniewn** [comment]
I have now updated this PR to capture all the relevant discussion in the NAP itself and update the governance. I have marked the PR ready for review from the @napari/core-devs team.

File: docs/community/governance.md
Comment (DragaDoncila) [inline docs/community/governance.md:94]: It is slightly unclear here whether consensus for admission to the steering council refers to a vote from the core developers/existing steering council and whether this consensus needs to be unanimous. In particular, can the steering council, or any of its members, veto admission of a nominee even when there is consensus among the core developers?

File: docs/community/governance.md
Comment (DragaDoncila) [inline docs/community/governance.md]:
```suggestion
diversity of voices. All deadlocked votes of the SC will be postponed until
```

File: docs/community/governance.md
Comment (DragaDoncila) [inline docs/community/governance.md:90]: I think "expected to resign" is slightly unclear. Who among core developers and/or existing SC (if anyone) has the ability to "expel" an SC member who is "not actively engaging"? Would this also be a consensus based decision?

It may be that this is not within scope for this document but it was unclear to me and I just wanted to highlight.

File: docs/community/governance.md
Comment (DragaDoncila) [inline docs/community/governance.md:135]: should a vote arise while the IFPAC is in the process of electing a new SC representative, is this vote postponed until that member is admitted? In other words, can an SC vote go through without a currently sitting IFPAC committee member

File: docs/naps/1-institutional-funding-partners.md
Comment (DragaDoncila) [inline docs/naps/1-institutional-funding-partners.md]:
```suggestion
The proposed implementation can be found in [PR 4458](https://github.com/napari/napari/pull/4458)
```

File: docs/naps/1-institutional-funding-partners.md
Comment (DragaDoncila) [inline docs/naps/1-institutional-funding-partners.md]:
```suggestion
NAP-1 does not prescribe the rules for the IFP Advisory Council, but does ask them to describe their
```

**Review by DragaDoncila — CHANGES_REQUESTED** [review]
Left some proofreading fixes and a few discussion questions/requests for clarification. It's shaping up very well though!

File: docs/community/governance.md
Comment (sofroniewn) [reply docs/community/governance.md:63]: fixed

File: docs/community/governance.md
Comment (sofroniewn) [reply docs/community/governance.md:90]: That's a valid question - it's language that is already existing in our governance, so not really in scope of the IFP change. For now I'd say if you have that concern you can elevate it to other members of the SC

File: docs/community/governance.md
Comment (sofroniewn) [reply docs/community/governance.md:94]: My understanding is that it would be a vote of the existing SC. The consensus would not need to be unanimous - but a simple majority of the SC, as is outlined in more detail here https://github.com/napari/napari/blob/main/docs/community/governance.md#decision-making-process.

File: docs/community/governance.md
Comment (sofroniewn) [reply docs/community/governance.md:135]: I would expect the SC and the IFPAC to work together in good faith to resolve this in a timely fashion

**Review by sofroniewn — COMMENTED** [review]
> Left some proofreading fixes and a few discussion questions/requests for clarification. It's shaping up very well though!

Thanks - i have now responded to your comments and incorporated some of your changes. Please let me know if you'd like more detailed responses/ more additions in the doc. Thanks!!

File: docs/community/governance.md
Comment (jni) [reply docs/community/governance.md:94]: Yes, unless otherwise specified, I would say all decisions are taken by the usual consensus-seeking process, which always ends up in the SC. At any rate I'll second your response to a later comment @sofroniewn — this NAP and PR don't actually change this language, so reviewing it is out of scope for the PR.

File: docs/naps/1-institutional-funding-partners.md
Comment (jni) [inline docs/naps/1-institutional-funding-partners.md:205]:
```suggestion
two IFP like seats. Additionally, research shows that collaboration and problem-solving
becomes harder for group sizes larger than five. Beyond that size, conversations often
become fragmented. A maximum size of five keeps the SC decision-making process
nimble.
```

**Review by jni — APPROVED** [review]
I've left a very minor suggestion for the discussion section. This is ready to go as far as I'm concerned!

**Review by AhmetCanSolak — APPROVED** [review]

**Review by DragaDoncila — APPROVED** [review]
Thank you for clarifying and adding more context around some of the language. Approving now 🎉

**Comment by jni** [comment]
Thank you everyone for all the useful discussions, tweaks and collaboration on this process! I'm gonna merge! :rocket:

**Pull request merged**
